scripts/prepdocs/files.py

```diff
@@ -0,0 +1,749 @@
+                    continue
+
+                yield path
+
+    async def list(self) -> AsyncGenerator[File, None]:
+        async with DataLakeServiceClient(
+            account_url=f"https://{self.data_lake_storage_account}.dfs.core.windows.net", credential=self.credential
+        ) as service_client, service_client.get_file_system_client(self.data_lake_filesystem) as filesystem_client:
+            async for path in self.list_paths():
+                temp_file_path = os.path.join(tempfile.gettempdir(), os.path.basename(path.name))
+                try:
+                    async with filesystem_client.get_file_client(path) as file_client:
+                        with open(temp_file_path, "wb") as temp_file:
+                            downloader = await file_client.download_file()
+                            await downloader.readinto(temp_file)
+
+                    # Parse out user ids and group ids
+                    acls = {"oids": [], "groups": []}
+                    # https://learn.microsoft.com/python/api/azure-storage-file-datalake/azure.storage.filedatalake.datalakefileclient?view=azure-python#azure-storage-filedatalake-datalakefileclient-get-access-control
+                    # Request ACLs as GUIDs
+                    access_control = await file_client.get_access_control(upn=False)
+                    acl_list = access_control["acl"]
+                    # https://learn.microsoft.com/azure/storage/blobs/data-lake-storage-access-control
+                    # ACL Format: user::rwx,group::r-x,other::r--,user:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx:r--
+                    acl_list = acl_list.split(",")
+                    for acl in acl_list:
+                        acl_parts: list = acl.split(":")
+                        if len(acl_parts) != 3:
+                            continue
+                        if len(acl_parts[1]) == 0:
+                            continue
+                        if acl_parts[0] == "user" and "r" in acl_parts[2]:
+                            acls["oids"].append(acl_parts[1])
+                        if acl_parts[0] == "group" and "r" in acl_parts[2]:
+                            acls["groups"].append(acl_parts[1])
+
+                    yield File(content=open(temp_file_path, "rb"), acls=acls)
+                except Exception as e:
+                    print(f"\tGot an error while reading {path.name} -> {e} --> skipping file")
+                    try:
+                        os.remove(temp_file_path)
+                    except Exception as e:
+                        print(f"\tGot an error while deleting {temp_file_path} -> {e}")
+
+
+class Page:
+    def __init__(self, page_num: int, offset: int, text: str):
+        self.page_num = page_num
+        self.offset = offset
+        self.text = text
+
+
+class SplitPage:
+    def __init__(self, page_num: int, text: str):
+        self.page_num = page_num
+        self.text = text
+
+
+class TextSplitter:
```

Comment:
I wonder if chunking strategies should go in their own module? Might be nice to reduce file size if that organization makes sense.

Reply:
Oh I see this is already a new module! Neat. Well, not sure if it should be split further, just an idea. You could look at -accelerator and https://github.com/microsoft/sample-app-aoai-chatGPT for inspiration, to imagine whether you'd want this in a separate file in a future world where we supported more file formats, for example.
Not strongly opinionated on this either way.

Reply:
I do think splitting this across files is the right move. Still trying to fix errors though first

Reply:
module-ified